run_tests.sh

```diff
@@ -0,0 +1,79 @@
+# Colors for output
+RED='\033[0;31m'
+GREEN='\033[0;32m'
+NC='\033[0m'
+
+docker-compose up -d
+if [ $? -ne 0 ] ; then
+    printf "${RED}docker-compose failed${NC}\n"
+    exit -1
+fi
+
+echo "Starting linting..."
+# Wait for lint
+LINT_TEST_EXIT_CODE=`docker wait lint`
+# Output lint logs
+docker logs lint
+# Check unit-tests output
+if [ -z ${LINT_TEST_EXIT_CODE+x} ] || [ "$LINT_TEST_EXIT_CODE" -ne 0 ] ; then
```

Comment:
what does this `+x` do?

Reply:
`${FOO+x}` evaluates to null if FOO is undefined and evaluates to "x" if FOO is defined. The check `[ -z ${FOO+x} ] ` is to check if the exit code is undefined. I think `[ -z ${FOO} ]` would fail to differentiate between FOO being undefined and FOO being an empty string